terminology/terminology_index.adoc

```diff
@@ -0,0 +1,207 @@
+are used interchangeably.It can be confusing, especially for newcomers.
+
+The goal of this section is to clarify these terms, so that we can speak the same language.
+
+==== Container Image
+
+Container image is a binary that includes all of the requirements for running a container, as well as metadata describing its needs and capabilities. You can think of it as a packaging technology. Containers only have access to resources defined in the image,
+unless you give the container additional access when creating it.
+
+==== Container
+
+A container is an instance of an image. Containers are an abstraction at the application layer that packages code and dependencies together.
+Multiple containers can run on the same machine and share the OS kernel with other containers, each running as an isolated processes in the user space.
+Containers take up less space than VMs (container images are typically tens of MBs in size), and start almost instantly.
+
+==== Repository
+
+When using the `docker` command, a repository is what is specified on the command line, not an image. In the following command, “fedora” is the repository.
+
+```
+docker pull fedora
+```
+
+This is actually expanded automatically to:
+
+```
+docker pull docker.io/library/fedora:latest
+```
+
+This can be confusing, and many people refer to this as an image or a container image. In fact, the docker images sub-command is what is used to list the locally available repositories. Conceptually, these repositories can be thought about as container images, but it’s important to realize that these repositories are actually made up of layers.
+
+When we specify the repository on the command line, the Docker daemon is doing some extra work for you. The Docker daemon (not the client tool) is configured with a list of servers to search. In our example above, the damone will search for the “fedora” repository on each of the configured servers.
+
+In the above command, only the repository name was specified, but it’s also possible to specify a full URL with the Docker client. To highlight this, let’s start with dissecting a full URL.
+
+```
+REGISTRY[:PORT]/NAMESPACE/REPOSITORY[:TAG]
+```
+
+The full URL is made up of a standard server name, a namespace, and optionally a tag. There are actually many permutations of how to specify a URL and as you explore the Docker ecosystem, you will find that many pieces are optional. The following commands are all valid and all pull some permutation of the same repository:
+
+```
+docker pull docker.io/library/fedora:latest
+docker pull docker.io/library/fedora
+docker pull library/fedora
+docker pull fedora
+```
+
+==== Image Layer
+
+Repositories are often referred to as images or container images, but actually they are made up of one or more layers.
+Image layers in a repository are connected together in a parent-child relationship. Each image layer represents changes between itself and the parent layer.
```

Comment:
Changes may suggest diff, but the fact is that layers are just pieces of the full puzzle. So maybe `Each image layer represents some pieces of the final container image`.